triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRuleSimulationLib.scala

```diff
@@ -799,13 +808,35 @@ final class TriggerRuleSimulationLib private (
         val submissions = updateStateSimulation.runWith(Sink.seq)
         val nextState = updateStateSimulation.toMat(Sink.ignore)(Keep.left).run()
 
-        submissions.zip(nextState)
+        submissions.map(_.map(_.value)).zip(nextState).map { case (requests, state) =>
+          (requests, ruleMetrics.getMetrics, state)
+        }
       }
     }
   }
 }
 
 object TriggerRuleSimulationLib {
+
+  type CommandsInFlight = Map[String, Seq[Command]]
+
+  final case class TriggerExperiment(
+      acsGen: Gen[Seq[CreatedEvent]],
+      userStateGen: Gen[SValue],
+      inFlightCmdGen: Gen[CommandsInFlight],
+      msgGen: Gen[TriggerMsg],
+  ) {
```

Comment:
This case class will form the back bone of testing under PR https://github.com/digital-asset/daml/pull/16508